app/Http/Controllers/Document/BulkDownloadController.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace KBox\Http\Controllers\Document;
+
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use KBox\Http\Controllers\Controller;
+use Zip;
+use KBox\Http\Requests\BulkDownloadRequest;
+use KBox\DocumentDescriptor;
+use Illuminate\Contracts\Auth\Guard as AuthGuard;
+use Illuminate\Support\Facades\Storage;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+use ZipStream\ZipStream;
+use Illuminate\Http\JsonResponse;
+
+use function Matrix\trace;
+
+class BulkDownloadController extends Controller
+{
+        /*
+    |--------------------------------------------------------------------------
+    | Bulk Operation on Documents and Groups Controller
+    |--------------------------------------------------------------------------
+    |
+    | handle the operation when something is performed on a multiple selection.
+    | To simply JS stuff
+    |
+    */
+
+    /**
+     * [$service description]
+     * @var \KBox\Documents\Services\DocumentsService
+     */
+    private $service = null;
+
+    /**
+     * Create a new controller instance.
+     *
+     * @return void
+     */
+    public function __construct(\KBox\Documents\Services\DocumentsService $adapterService)
+    { 
+        $this->middleware('auth');
+
+        $this->service = $adapterService;
+    }
+
+    
+    //
+    public function buildzip(AuthGuard $auth, BulkDownloadRequest $request){
+
+        try {
+            \Log::info('Bulk  Download', ['params' => $request->all()]);
+            
+            //$user = $auth->user();
+            $docs = $request->input('documents', []);
```

Comment:
Input should be [validated](https://laravel.com/docs/7.x/validation#quick-writing-the-validation-logic). We can just assume to refuse to create the zip if the documents list contains documents that are not accessible by the user